client/memconfig.go

```diff
@@ -0,0 +1,153 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const (
+	memConfigEndpoint = "/memstore/config"
+)
+
+// MemConfig ... contains properties that are used to configure the MemStore's behavior.
+// this is copied directly from /store/generated_key/memstore/memconfig. 
+// importing the struct isn't possible since it'd create cyclic dependency loop 
+// with core proxy's go.mod
+type MemConfig struct {
+	MaxBlobSizeBytes uint64
+	BlobExpiration   time.Duration
+	PutLatency time.Duration
+	GetLatency time.Duration
+	PutReturnsFailoverError bool
+}
+
+// MarshalJSON implements custom JSON marshaling for Config.
+// This is needed because time.Duration is serialized to nanoseconds,
+// which is hard to read.
+func (c MemConfig) MarshalJSON() ([]byte, error) {
+	return json.Marshal(intermediaryCfg{
+		MaxBlobSizeBytes:        c.MaxBlobSizeBytes,
+		BlobExpiration:          c.BlobExpiration.String(),
+		PutLatency:              c.PutLatency.String(),
+		GetLatency:              c.GetLatency.String(),
+		PutReturnsFailoverError: c.PutReturnsFailoverError,
+	})
+}
+
+
+// intermediaryCfg ... used for decoding into a less rich type before
+// translating to a structured MemConfig
+type intermediaryCfg struct {
+	MaxBlobSizeBytes        uint64
+	BlobExpiration          string
+	PutLatency              string
+	GetLatency              string
+	PutReturnsFailoverError bool
+}
+
+// IntoMemConfig ... converts an intermediary config into a memconfig 
+// with structured type definitions
+func (cfg *intermediaryCfg) IntoMemConfig() (*MemConfig, error) {
+	getLatency, err := time.ParseDuration(cfg.GetLatency)
```

Comment:
Can't we just implement UnmarshalJSON directly on MemConfig the same way we implemented the custom MarshalJSON?

Reply:
we'd still need to cast into `time.Duration` right?

Reply:
yes but the point is that this intermediaryCfg would only be defined inside the UnmarshalJSON, and hidden as an implementation detail. Whereas the way its currently structured it feels like it's something important, but it really doesn't outside of unmarshalling.

Reply:
eh I don't see any great benefits with this approach over the existing implementation